tests/unit/test_utils.py

```diff
@@ -422,6 +422,54 @@ def test_update_exc(self):
         ret = utils.update_exc(err, msg2, before=False, separator=sep)
         self.assertEqual(ret.message, exp)
 
+    def test_read_in_chunks(self):
+        # create junk file to test size.
+        source_file = "source_file.dat"
```

Comment:
Instead of hard-coding the temporary file names, why not use [TemporaryFile](https://docs.python.org/2.7/library/tempfile.html#tempfile.TemporaryFile)?